src/mpid/ch4/src/ch4_coll.h

```diff
@@ -13,112 +13,148 @@
 
 #include "ch4_impl.h"
 #include "ch4r_proc.h"
-#include "ch4_coll_select.h"
+#include "ch4_coll_impl.h"
 
 MPL_STATIC_INLINE_PREFIX int MPID_Barrier(MPIR_Comm * comm, MPIR_Errflag_t * errflag)
 {
     int mpi_errno = MPI_SUCCESS;
-    const MPIDI_coll_algo_container_t *ch4_algo_parameters_container = NULL;
+    const MPIDI_coll_algo_container_t *cnt = NULL;
+
+    MPIR_Csel_coll_sig_s coll_sig = {
+        .coll_type = MPIR_CSEL_COLL_TYPE__BARRIER,
+        .comm_ptr = comm,
+    };
 
     MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_MPID_BARRIER);
     MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPID_BARRIER);
 
-    ch4_algo_parameters_container = MPIDI_Barrier_select(comm, errflag);
+    cnt = MPIR_Csel_search(MPIDI_COMM(comm, csel), coll_sig);
 
-    switch (ch4_algo_parameters_container->id) {
-        case MPIDI_Barrier_intra_composition_alpha_id:
-            mpi_errno =
-                MPIDI_Barrier_intra_composition_alpha(comm, errflag, ch4_algo_parameters_container);
+    if (cnt == NULL) {
+        mpi_errno = MPIR_Barrier_impl(comm, errflag);
+        goto fn_exit;
+    }
+
+    switch (cnt->id) {
+        case MPIDI_CSEL_CONTAINER_TYPE__COMPOSITION__MPIDI_Barrier_intra_composition_alpha:
+            mpi_errno = MPIDI_Barrier_intra_composition_alpha(comm, errflag);
             break;
-        case MPIDI_Barrier_intra_composition_beta_id:
-            mpi_errno =
-                MPIDI_Barrier_intra_composition_beta(comm, errflag, ch4_algo_parameters_container);
+        case MPIDI_CSEL_CONTAINER_TYPE__COMPOSITION__MPIDI_Barrier_intra_composition_beta:
+            mpi_errno = MPIDI_Barrier_intra_composition_beta(comm, errflag);
             break;
         default:
-            mpi_errno = MPIR_Barrier_impl(comm, errflag);
-            break;
+            MPIR_Assert(0);
     }
 
+  fn_exit:
     MPIR_FUNC_VERBOSE_EXIT(MPID_STATE_MPID_BARRIER);
     return mpi_errno;
+  fn_fail:
```

Comment:
This is still an unused label. We're going to get new warnings.

Reply:
I think the same is true throughout this file.

Reply:
Fixed.